test/utils/Partition_One_Test.cpp

```diff
@@ -0,0 +1,153 @@
+
+    std::vector<std::string> catchment_subset_ids;
+    std::vector<std::string> nexus_subset_ids;
+
+    std::unordered_set<std::string> catchment_ids;
+    std::unordered_set<std::string> nexus_ids;
+
+    Partition_One partition_one;
+    PartitionData partition_data;
+
+    PartitionOneTest() {} 
+
+    ~PartitionOneTest() override {}
+
+    std::string file_search(const std::vector<std::string> &parent_dir_options, const std::string& file_basename)
+    {
+        // Build vector of names by building combinations of the path and basename options
+        std::vector<std::string> name_combinations;
+
+        // Build so that all path names are tried for given basename before trying a different basename option
+        for (auto & path_option : parent_dir_options)
+            name_combinations.push_back(path_option + file_basename);
+
+        return utils::FileChecker::find_first_readable(name_combinations);
+    }
+
+    void read_file_generate_partition_data()
+    {
+        const std::string file_path = file_search(hydro_fabric_paths, "catchment_data.geojson");
+        catchment_collection = geojson::read(file_path, catchment_subset_ids);
+
+        for(auto& feature: *catchment_collection)
+        {
+            std::string cat_id = feature->get_id();
+            partition_data.catchment_ids.emplace(cat_id);
+            std::string nex_id = feature->get_property("toid").as_string();
+            partition_data.nexus_ids.emplace(nex_id);
+        }
+    }
+
+    void SetUp() override;
+
+    void TearDown() override;
+
+    void setupArbitraryExampleCase();
+
+};
+
+void PartitionOneTest::SetUp() {
+    setupArbitraryExampleCase();
+}
+
+void PartitionOneTest::TearDown() {
+
+}
+
+void PartitionOneTest::setupArbitraryExampleCase() {
+    hydro_fabric_paths = {
+        "data/",
+        "./data/",
+        "../data/",
+        "../../data/",
+
+    };
+}
+
+TEST_F(PartitionOneTest, TestPartitionData_1a)
+{
+    read_file_generate_partition_data();
+
+    partition_one.generate_partition(catchment_collection);
+    PartitionData data_struct = partition_one.partition_data;
+    catchment_ids = data_struct.catchment_ids;
+
+    //check catchment partition
+    std::vector<std::string> cat_id_vec;
+    // convert unordered_set to vector
+    for (const auto& id: catchment_ids) {
+        cat_id_vec.push_back(id);
+    }
+
+    //sort ids
+    std::sort(cat_id_vec.begin(), cat_id_vec.end());
+    //create set of unique ids
+    std::set<std::string> unique(cat_id_vec.begin(), cat_id_vec.end());
+    std::set<std::string> duplicates;
+    //use set difference to identify all duplicates
+    std::set_difference(cat_id_vec.begin(), cat_id_vec.end(), unique.begin(), unique.end(), std::inserter(duplicates, duplicates.end()));
+
+    for( auto& id: duplicates){
+        if (!id.empty()) {
+            std::cout << "duplicates string set is not empty" << std::endl;
+            break;
+        }
+    }
```

Comment:
```suggestion
    for (auto& id: duplicates) {
        std::cout << "duplicates string set contains " << id << std::endl;
    }
```
The `if` condition is testing for `duplicates` containing an element that is an empty string (`id` is a `std::string`. That makes no sense. If `duplicates` contains any elements at all, the error message applies, and it should print the element(s) found.

The exact same concern applies to test 1b below.

Reply:
The if condition tests for duplicates that's not empty by the ```negation operator```. So if that condition holds, we print out a message that says there are duplicates. Then break out of the loop so we don't need to print out a lot of lines if there are many in the duplicates.